cpp/include/cuspatial/error.hpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cusparse.h>
```

Comment:
This causes every cuspatial file to include a ton of code it doesn't use.  Instead, just forward declare `cuSparseStatus_t`. Better yet, move all the cusparse stuff to a separate utility header.

Reply:
When you say move to a separate header, do you mean forward-declare everything we need from `cusparse.h`, or just `cuSparseStatus_t`?

Reply:
I think if you make a separate cusparse_error.hpp you can include cusparse.h from it and you don't have to forward decl anything.  This just saves all of cuspatial having to include cusparse.h except for the one file where we use it.